File: common/com.tibco.cep.studio.common/src/com/tibco/cep/webstudio/model/rule/instance/impl/MultiFilterImpl.java
package com.tibco.cep.webstudio.model.rule.instance.impl;

import com.tibco.cep.webstudio.model.rule.instance.BuilderSubClause;
import com.tibco.cep.webstudio.model.rule.instance.IInstanceChangedEvent;
import com.tibco.cep.webstudio.model.rule.instance.InstanceChangedEvent;
import com.tibco.cep.webstudio.model.rule.instance.MultiFilter;

public class MultiFilterImpl extends RuleTemplateInstanceObject implements MultiFilter {

	/**
	 * 
	 */
	private static final long serialVersionUID = -5223570507083214135L;
	private String matchType;
	private BuilderSubClause subClause;

	public MultiFilterImpl() {
		this.matchType = "Match Any";
		this.subClause = new BuilderSubClauseImpl();
	}
	
	public MultiFilterImpl(String matchType) {
		this.matchType = matchType;
		this.subClause = new BuilderSubClauseImpl();
	}

	@Override
	public String getMatchType() {
		return this.matchType;
	}

	@Override
	public void setMatchType(String value) {
		this.matchType = value;
		InstanceChangedEvent changeEvent = new InstanceChangedEvent(IInstanceChangedEvent.CHANGED, MATCH_TYPE_FEATURE_ID, this, value);
		fireInstanceChanged(changeEvent);
	}

	@Override
	public BuilderSubClause getSubClause() {
		return this.subClause;
	}

	@Override
	public void setSubClause(BuilderSubClause value) {
		this.subClause = value;
		InstanceChangedEvent changeEvent = new InstanceChangedEvent(IInstanceChangedEvent.CHANGED, SUB_CLAUSE_FEATURE_ID, this, value);
		fireInstanceChanged(changeEvent);
	}
	
	@Override
	public String getFilterId() {
		return filterId;
	}
	
	@Override
	public void setFilterId(String filterId) {
		this.filterId = filterId;
	}
}

File: designtime/webstudio/ui/src/app/editors/rule-template-instance-builder/MultiFilterImpl.ts
/**
 * @Author: Rahil Khera
 * @Date:   2017-04-20T10:04:16+05:30
 * @Last modified by:   Rahil Khera
 * @Last modified time: 2017-04-20T10:04:22+05:30
 */

import { BuilderSubClauseImpl } from './BuilderSubClauseImpl';
import { Filter } from './Filter';

export type MatchType = 'Match All' | 'Match Any' | 'Match None';

export class MultiFilterImpl extends Filter {

  private matchType: MatchType = undefined;
  private builderSubClause: BuilderSubClauseImpl = new BuilderSubClauseImpl(); // filter

  constructor(matchType?: MatchType) {
    super();
    this.setFilterId(this.getFilterId());
    this.matchType = matchType == null ? 'Match All' : matchType;
    this.builderSubClause = new BuilderSubClauseImpl();
  }

  getMatchType(): MatchType {
    return this.matchType;
  }

  setMatchType(matchType: MatchType) {
    this.matchType = matchType;
  }

  getBuilderSubClause(): BuilderSubClauseImpl {
    return this.builderSubClause;
  }

  setBuilderSubClause(value: BuilderSubClauseImpl) {
    this.builderSubClause = value;
  }
}
